src/Orleans/Serialization/SerializationManager.cs

```diff
@@ -137,30 +137,26 @@ public static object GetUninitializedObjectWithFormatterServices(Type type)
 
         #region Static initialization
 
-        public static void InitializeForTesting(List<TypeInfo> serializationProviders = null)
+        public static void InitializeForTesting(List<TypeInfo> serializationProviders = null, IExternalSerializer fallback = null)
         {
             try
             {
                 RegisterBuiltInSerializers();
                 BufferPool.InitGlobalBufferPool(new MessagingConfiguration(false));
                 RegisterSerializationProviders(serializationProviders);
                 AssemblyProcessor.Initialize();
-                fallbackSerializer = GetFallbackSerializer();
+                fallbackSerializer = fallback ?? GetFallbackSerializer();
             }
             catch (ReflectionTypeLoadException ex)
             {
                 throw ex.Flatten();
             }
         }
 
-        internal static void Initialize(List<TypeInfo> serializationProviders)
+        internal static void Initialize(List<TypeInfo> serializationProviders, IExternalSerializer fallback = null)
```

Comment:
Good question. At the moment I added a commit atop this to select the fallback serializer even on full .NET, but if you prefer a config option then how about I add one which is similar to the `SerializationProviders` config & is represented by a `TypeInfo`?